### PR TITLE
Split the payload decoding and session injection so payload decoding can be asynchronously extended

### PR DESCRIPTION
**Pull request by HughePaul**

### CODE REVIEW

File: injection/session-injection.js [hunk 1]
```diff
@@ -145,9 +145,7 @@ class SessionInjection {
         deepCloneMerge.extend(req.session, rawSessionValues);
     }
 
-    middlewareHandler(req, res, next) {
-        debug('middlewareHandler');
-
+    middlewareDecodePayload(req, res, next) {
         let payload;
```
Comment (merlinc) [inline injection/session-injection.js]: Add a `debug('middlewareDecodePayload')` line for consistency

File: test/validation/spec.postcode-validation.js [hunk 2]
```diff
@@ -20,8 +20,8 @@ describe('Postcode validation', () => {
     });
 
     describeUnlessCoverage('Full postcode test - loads full UK postcode database, may take some time', function () {
-        this.timeout(40000);
-        this.slow(20000);
+        this.timeout(60000);
```
Comment (merlinc) [inline test/validation/spec.postcode-validation.js]: Do we want to have a separate MR for removing the postcode validation? We don't use it anymore, and increasing the timeouts is now getting silly.

File: injection/session-injection.js (hunk 1 above)
Comment (HughePaul) [reply injection/session-injection.js]: I'll add this in
